### ISSUE TITLE
Thoughts on using flag `O_CLOEXEC` for unix source sets?

### ISSUE THREAD

**Issue by 05nelsonm** [square/okio#1447]
Wanted to pop in and ask about the possibility of always opening descriptors with the `O_CLOEXEC` flag, or utilizing the `e` flag with `fopen`.

**Comment by swankjesse** [comment]
Yeah, seems quite reasonable. Do you have a specific use case? I’m curious about forking and execing in Kotlin/Native and haven’t seen many examples.

**Comment by swankjesse** [comment]
Please send a PR!

**Comment by 05nelsonm** [comment]
> Do you have a specific use case?

Not really, just thought it'd be an easy win for everyone.

> I’m curious about forking and execing in Kotlin/Native and haven’t seen many examples

https://github.com/05nelsonm/kmp-process/blob/master/library/process/src/unixMain/kotlin/io/matthewnelson/kmp/process/internal/PlatformBuilder.kt
